components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/Connection.kt

```diff
@@ -22,14 +29,126 @@ const val DB_NAME = "places.sqlite"
  *
  * Writer is always the same, as guaranteed by [PlacesApi].
  */
-interface Connection : Closeable {
+internal interface Connection : Closeable {
     fun reader(): PlacesReaderConnection
     fun writer(): PlacesWriterConnection
 
     // Until we get a real SyncManager in application-services libraries, we'll have to live with this
     // strange split that doesn't quite map all that well to our internal storage model.
     fun syncHistory(syncInfo: SyncAuthInfo)
     fun syncBookmarks(syncInfo: SyncAuthInfo)
+
+    // These are implemented as default methods on `Connection` instead of
+    // `RustPlacesConnection` to make testing easier.
+    @Suppress("ComplexMethod", "NestedBlockDepth")
+    fun assembleHistoryPing(ping: SyncTelemetryPing) {
+        if (!Glean.getUploadEnabled()) {
```

Comment:
Are you doing this for performance reasons? Because, on our end, accumulating to our metrics is a no-op if upload is disabled. I'd consider dropping this for now and just doing that if there's any performance issue.

Reply:
Yep, it was just to save on unnecessary work, but it's a bit silly. Dropped!